### ISSUE TITLE
null parameter to workflow step

### ISSUE THREAD

**Issue by patriknw** [lightbend/kalix-jvm-sdk#1919]
We didn't have the need of a command parameter for the transitionTo, and therefore tried `null` and `Void.class, __ -> {`

It failed with a serialization error message. Would be better to reject null immediately with a clear error message. Alternatively support null.

**Comment by aludwiko** [comment]
Can't you just use transition without the input: 
![image](https://github.com/lightbend/kalix-jvm-sdk/assets/2954053/85d9f32d-1b9f-44cc-8581-7eb01c688778)

**Comment by patriknw** [comment]
Yes, we found that overload later. Would anyway be good to have a fail fast check for `null`.
